lib/FieldType/Mapper/FormTypeBasedFieldValueFormMapper.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * This file is part of the eZ RepositoryForms package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+namespace EzSystems\RepositoryForms\FieldType\Mapper;
+
+use eZ\Publish\API\Repository\FieldTypeService;
+use EzSystems\RepositoryForms\Data\Content\FieldData;
+use EzSystems\RepositoryForms\FieldType\DataTransformer\FieldValueTransformer;
+use EzSystems\RepositoryForms\FieldType\FieldValueFormMapperInterface;
+use Symfony\Component\Form\FormInterface;
+
+/**
+ * Generic FieldValueFormMapper that uses a named FormType to generate a FieldValue Form.
+ * 
+ * The FieldValueTransformer is used as a model transformer.
+ * 
+ * Example in YAML service definition:
+ * ```
+ * ezrepoforms.field_type.form_mapper.ezuser:
+ *   class: "%ezrepoforms.field_type.form_mapper.form_type_based.class%"
+ *   parent: ezrepoforms.field_type.form_mapper.form_type_based
+ *   tags:
+ *     - { name: "ez.formMapper.fieldValue", fieldType: "ezuser" }
+ *   calls:
+ *     - [setFormType, ["ezuser"]]
+ * ```
+ */
+final class FormTypeBasedFieldValueFormMapper implements FieldValueFormMapperInterface
+{
+    /**
+     * The FormType used by the mapper. Example: 'text'.
+     * @var string
+     */
+    private $formType;
+
+    /**
+     * @var \eZ\Publish\API\Repository\FieldTypeService
+     */
+    private $fieldTypeService;
+
+    public function __construct(FieldTypeService $fieldTypeService)
+    {
+        $this->fieldTypeService = $fieldTypeService;
+    }
+
+    public function setFormType($formType)
+    {
+        $this->formType = $formType;
+    }
+
+    /**
+     * "Maps" Field form to current FieldType.
```

Comment:
Why the quotation marks? (If `map` isn't really the right word here, then it shouldn't be used in the method name either.) I see the interfaces also quote it, looks wrong there too :)